File: api-aggregator/src/main/java/com/github/alexeysol/geekregimeapiaggregator/shared/utils/sources/ApiUsersSource.java
package com.github.alexeysol.geekregimeapiaggregator.shared.utils.sources;

import com.github.alexeysol.geekregimeapicommons.models.sources.ApiPath;
import com.github.alexeysol.geekregimeapicommons.models.sources.BaseUrl;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class ApiUsersSource implements ApiPath, BaseUrl {
    @Value("${api-users.base-url}")
    private String baseUrl;

    @Value("${api-users.prefix}")
    private String apiPrefix;

    @Value("${api-users.resource}")
    private String resource;

    public String getBaseUrl() {
        return baseUrl;
    }

    public String getApiPath(int version) {
        return String.format("/%s/v%d/%s", apiPrefix, version, resource);
    }
}

File: src/main/kotlin/com/github/alexeysol/geekregimeapiusers/utils/sources/ApiUsersSource.kt
package com.github.alexeysol.geekregimeapiusers.utils.sources

import com.github.alexeysol.geekregimeapicommons.models.ApiPath
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
data class ApiUsersSource (
    private val PRODUCTION_PROFILE: String? = "prod",

    @Value("\${spring.profiles.active}")
    private val activeProfile: String? = null,

    @Value("\${api-users.prefix}")
    private val apiPrefix: String? = null,

    @Value("\${api-users.resource}")
    val resource: String? = null,
) : ApiPath {
    fun isProduction(): Boolean = PRODUCTION_PROFILE == activeProfile

    override fun getApiPath(version: Int): String = formatApiPath(apiPrefix, version, resource)
}
